doc/snmp-schema-addition.md

```diff
@@ -0,0 +1,83 @@
+# SONiC SNMP TABLE schema proposal #
+
+## Scope of the change ##
+
+Currently SNMP configuration is managed from a mix of yaml files and DB based ACLs, we propose to integrate the SMP configs into an SNMP table in the DB.  
+The following document proposes the database Schema in Json format and the list of required changes to the code.
+
+## Current configurtions for SNMP ##
+sonic-buidimage and sonic-snmpagent depend on the following files:
+1. */etc/sonic/snmp.yaml*
+    defines 2 keys:  
+        *snmp_rocommunity:*  
+        *snmp_location:*  
+   this file is consumed in the snmp docker container by */usr/share/sonic/templates/snmpd.conf.j2* and */usr/bin/snmpd-config-updater*.  
+2. */etc/sonic/config_db.json* or */etc/sonic/minigraph.xml* for ACL definitions 'SNMP_ACL'
+   this file is consumed by */usr/bin/snmpd-config-updater*, */usr/share/sonic/templates/alias_map.j2*, */usr/share/sonic/templates/sysDescription.j2*.
+
+In the end this is used to produce */etc/snmp/snmpd.conf*.
+
+## Limits and incentives to change ##
+- This goes against the principle of having the config DB be the central repository of configuration data.
+- Imposes the provisioning of multiple static files.
+- the snmpContact is hardcoded in the */usr/share/sonic/templates/snmpd.conf.j2* file.
+- pass-throughs are hard-coded.
+
+## Proposal: integrate the SNMP configuration into config DB ##
+### Proposed Schema ###
+
+```
+"SNMP": {
+    "location": LOCATION_STRING,
+    "contact": CONTACT_STRING,
+    "v2c": {
+        COMMUNITY_STRING: {
```

Comment:
>COMMUNITY_STRING [](start = 8, length = 16)

Is it true that COMMUNITY_STRING is not related to v2c?

Reply:
I don't understand the comment
it is my understanding that in v1 and v2c the community is a 'password' and that in v3 this 'password' is replaced by user authentication+encryption key, so in this contect COMMUNITY_STRING should be part of v2c configuration only

Reply:
I found "SNMPv3 also uses community strings, but allows for secure authentication and communication between SNMP manager and agent"
[https://en.wikipedia.org/wiki/Simple_Network_Management_Protocol](https://en.wikipedia.org/wiki/Simple_Network_Management_Protocol).
I am ok with the version string here.

---
In reply to: [211706220](https://github.com/Azure/SONiC/pull/231#discussion_r211706220) [](ancestors = 211706220)